File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.py
class Solution:
    def maximumElementAfterDecrementingAndRearranging(self, arr: List[int]) -> int:
        temp=sorted(arr)
        if temp[0]!=1: temp[0] = 1
        n = len(arr)
        for i in range(1,n):
            if temp[i]-temp[i-1]<=1: continue
            else: temp[i] = temp[i-1]+1
        currMx = max(temp)
        return currMx

File: 1956-maximum-element-after-decreasing-and-rearranging/maximum-element-after-decreasing-and-rearranging.cpp
class Solution {
public:
    int maximumElementAfterDecrementingAndRearranging(vector<int>& arr) {
        sort(arr.begin(),arr.end());
        arr[0] = arr[0] == 1 ? arr[0] : 1;
        for(int i =1; i<arr.size();i++){
            arr[i] = arr[i]-arr[i-1]<=1 ? arr[i] : arr[i-1]+1;
        }
        int mx = 0;for(int& n: arr){mx = max(mx,n);}
        return mx;
    }
};